cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/OperationsDashboardHttpHandler.java

```diff
@@ -103,18 +104,48 @@ public void readDashboardDetail(FullHttpRequest request, HttpResponder responder
       throw new BadRequestException("'namespace' cannot be empty, please provide at least one namespace.");
     }
     long endTimeSecs = startTimeSecs + durationTimeSecs;
+
+    // Get current running timestamp.
+    long currentTimeInSeconds = TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis());
+
+    // If the Current timestamp falls within the window, get all programs that are currently running.
```

Comment:
should there be a check that it's in the window before making this call?

Reply:
Oh, good catch. I switched my approach once I took a better look at the UI and discussed the requirements with product.

I'll remove this comment.

Reply:
I've updated the comment to better reflect what the code is doing.